### PR TITLE
Language Options

### PR DESCRIPTION
**Pull request by gaelyn**
## Summary by CodeRabbit

- **New Features**
    - Introduced language management with the `LanguageProvider` and `LanguageSwitcher` components.
    - Enhanced language support in various services and components, allowing users to select and update language preferences across the platform.
    - Added `language` parameter handling in API routes for creating and visiting spaces.

### CODE REVIEW

File: app/(authenticated)/layout.tsx [hunk 1]
```diff
@@ -31,8 +33,12 @@ export default async function Layout({
       >
         <NavItems />
       </div>
-
-      {children}
+      <LanguageProvider>
+        {children}
+        <div className="fixed top-4 right-4 z-50">
+          <LanguageSwitcher />
+        </div>
```
Comment (johnmosesman) [inline app/(authenticated)/layout.tsx:40]: Instead of fixed positioning can we make this a row at the top and justify-end?

Comment (johnmosesman) [reply app/(authenticated)/layout.tsx:40]: image
like in hcm

Comment (gaelyn) [reply app/(authenticated)/layout.tsx:40]: This seems harder than it should be with the way the layouts are. I'll keep working at it.

Comment (gaelyn) [reply app/(authenticated)/layout.tsx:40]: mainly because of the different bg colors on home page vs the other pages.

**Comment by gaelyn** [comment]
Screenshot 2024-04-25 at 2 36 14 PM
Screenshot 2024-04-25 at 2 36 40 PM
Screenshot 2024-04-25 at 2 37 14 PM

**Review by johnmosesman — APPROVED** [review]

**Pull request merged**